src/main/java/frc/robot/subsystems/Swerve.java

```diff
@@ -0,0 +1,32 @@
+package frc.robot.subsystems;
+
+// STEP 2: Add the epilogue @Logged annotation
+// STEP 2: Extend Swerve with SubsystemBase
+public class Swerve {
+
+  // STEP 5: Add a prviate Pose2d estimatedPose. This should automatically get picked up for logging
+  // by epilogue
+
+  public Swerve() {
+
+    // STEP 5: Initialize the Pose2d. An empty constructor will set the robot at (0,0) on the field,
+    // facing to the right.
+
+  }
+
+  /**
+   * Step 3: Public Method driveFieldRelative that takes field-relative inputs
```

Comment:
Method should be private, there should then be a public command factory that turns the method into a command

Reply:
@JakeStew81  I went back and forth on this one a bit for the workshop. 
PathPlanner expects a public method that accepts chassis speeds.
For simplicity, I went with having a single interface for driving. This method that takes joystick inputs in Lesson 1 gets refactored to chassis speeds in Lesson 4. This lesson should probably have instructions for testing running with PathPlanner.

Or... does the workshop need to include PathPlanner? The goal of the workshop is the swerve itself. If a student understands how a swerve drivetrain works and how to control it, they should be able to implement the PathPlanner interface. 

Given that, I can remove PathPlanner from the lesson and change the swerve api to the command factory style.

Reply:
You can configure the auto builder inside of swerve, which gets around the need for the method to be public. As far as needing pathplanner in the first place, I would agree that if a student knows the ins & outs of a swerve drivetrain, they should be able to use pathplanner fairly easily.